### ISSUE TITLE
Function of `label_len` in data_loader

### ISSUE THREAD

**Issue by rajatsen91** [cure-lab/LTSF-Linear#18]
Hello,

I do not fully understand the purpose of `label_len` in https://github.com/cure-lab/LTSF-Linear/blob/726695f784ebae2b4a7ff9bd0b34fa5a2c477b00/data_provider/data_loader.py#L276

Does this not lead to an overlap between `seq_x` and `seq_y`? Thanks in advance.

**Comment by mixiancmx** [comment]
Hi, when the data loader reads data, label_len is the length of the overlap between seq_x and seq_y. Seq_y has a length of label_len + pred_len. Transformer-based methods use the overlap part as the input of their decoder, i.e seq_y[:,:label_len,:].  seq_y[:,-pred_len:,:] will be used as ground truth. 
Our Linear models don't need this hyperparameter, but you need to keep it smaller than seq_len so that the data loader can run correctly.

**Comment by peterochek** [comment]
> Hi, when the data loader reads data, label_len is the length of the overlap between seq_x and seq_y. Seq_y has a length of label_len + pred_len. Transformer-based methods use the overlap part as the input of their decoder, i.e seq_y[:,:label_len,:]. seq_y[:,-pred_len:,:] will be used as ground truth. Our Linear models don't need this hyperparameter, but you need to keep it smaller than seq_len so that the data loader can run correctly.

I've read all the code, i don't see the point, where use explicitly don't use label_len. It seems to me that by default all Linear models are trained with overlap. In ```Exp_Main``` deafult is 48, then it is passed to ```args```, then ```DatasetCustom``` initialised with ```label_len = 48```. After it is used in ```__getitem__```

**Comment by chendingyan** [comment]
Does it mean that label_len hyperparameter for DLinear/NLinear model should be zero? So there is no overlap between seq_x and seq_y？ @mixiancmx

**Comment by thuliu-yt16** [comment]
I think DLinear and NLinear are not trained with overlap. For linear models, the input is `seq_x` only and the output has a length of `pred_len`. In the loss calculation (see `exp/exp_main.py` Line 166~168), both the output and the ground truth `seq_y` are truncated to just the last `pred_len` datapoints. So actually `seq_y[:-pred_len]` is not used in linear model training.

**Comment by peterochek** [comment]
Now I see. It was quite complicated to understand it. Thanks!

**Comment by esnvidia** [comment]
@mixiancmx So for the most amount of context passed to the model should `seq_len == label_len + pred_len`?
So for:

```
index = 0
seq_len = 100
label_len = 99
pred_len = 1
```

Then
```
s_begin -> 0
s_end = s_begin + seq_len  # -> 100
r_begin = s_end - label_len # -> 1
r_end = r_begin + label_len + pred_len # -> 1+99+1 = 101. 
```
so  (s_begin, s_end) -> (0,100)
and (r_begin, r_end) -> (1,101)

Is this the correct way to understand this in the Nowcasting case?
